lms/djangoapps/commerce/views.py

```diff
@@ -0,0 +1,179 @@
+            msg = 'E-Commerce API not setup. Enrolled {} in {} directly.'.format(user.username, unicode(course_key))
+            log.debug(msg)
+            return DetailResponse(msg)
+
+        # Default to honor mode. In the future we may expand this view to support additional modes.
+        mode = CourseMode.DEFAULT_MODE_SLUG
+        course_modes = CourseMode.objects.filter(course_id=course_key, mode_slug=mode, sku__isnull=False)
+
+        # If there are no course modes with SKUs, enroll the user without contacting the external API.
+        if not course_modes.exists():
+            msg = _('The %(enrollment_mode)s mode for %(course_id)s does not have a SKU. '
+                    'Enrolling %(username)s directly.')
+            msg = msg % {
+                'enrollment_mode': mode,
+                'course_id': unicode(course_key),
+                'username': user.username
+            }
+            log.debug(msg)
+            self._enroll(course_key, user)
+            return DetailResponse(msg)
+
+        # Contact external API
+        headers = {
+            'Content-Type': 'application/json',
+            'Authorization': 'JWT {}'.format(self._get_jwt(user))
+        }
+
+        url = '{}/order'.format(ecommerce_api_url.strip('/'))
+
+        try:
+            response = requests.post(url, data={'sku': course_modes[0].sku}, headers=headers)
+        except Exception as ex:  # pylint: disable=broad-except
+            log.exception('Call to E-Commerce API failed: %s.', ex.message)
+            return ApiErrorResponse()
+
+        status_code = response.status_code
+
+        try:
+            data = response.json()
+        except JSONDecodeError as ex:
+            log.error('E-Commerce API response is not valid JSON.')
+            return ApiErrorResponse()
+
+        if status_code == HTTP_200_OK:
+            order_number = data.get('number')
+            order_status = data.get('status')
+            if order_status == OrderStatus.COMPLETE:
+                msg = _('Order %(order_number)s was completed.') % {'order_number': order_number}
+                log.debug(msg)
+                return DetailResponse(msg)
+            else:
+                # TODO Before this functionality is fully rolled-out, this branch should be updated to NOT enroll the
+                # user. Enrollments must be initiated by the E-Commerce API only.
+                self._enroll(course_key, user)
+                msg = 'Order %(order_number)s was received with %(status)s status. Expected %(complete_status)s. ' \
+                      'User %(username)s was enrolled in %(course_id)s by LMS.'
+                msg_kwargs = {
+                    'order_number': order_number,
+                    'status': order_status,
+                    'complete_status': OrderStatus.COMPLETE,
+                    'username': user.username,
+                    'course_id': unicode(course_key),
+                }
+                log.error(msg, msg_kwargs)
```

Comment:
Didn't we agree that if a request to the ecommerce service fails, the behavior currently in place should be preserved? I don't know that we want to be enrolling the user by contacting the enrollment API directly in the event of a fulfillment failure.

That said, I see your comment, and I'm okay with this as a temporary measure meant only for the first iteration.